ipstools/IPDatabase.py

```
@@ -295,24 +302,95 @@ def generate_deps_tree(self, verbose=False):
+        else:
+            return None
+
+
+
+    def resolve_deps_conflicts(self, verbose=False, use_semver_resolution=False):
         """Resolves the IP dependency conflicts in the IP hierarchical flow.                    
 
             :param verbose:             If true, prints all information on the dependencies that are being fetched.
+            :param use_semver_resolution: bool If true, try to auto-resolve conflicts considering semantic versionsing rules.
             :type  verbose: bool
 
             :returns: `list` -- the final list of IPs after resolving all conflicts.
 
         This function resolves conflicts between IPs in a hierarchical flow by calling for the user's intervention.
         """
 
+        if use_semver_resolution:
+            if semver_available:
+                # In order to resolve transitive conflicts (i.e. conflicts of conflicts) automatically, we need
+                # to resolve conflicts using semver and re-evaluate if this fixed any other conflicts until there
+                # are no conflicts left or we hit a steady state
+                ip_tree_updated = True # this variable keeps track whether we update the ip_tree in the current iteration
+                                        # and thus allows us to detect steady states.
+                while ip_tree_updated:
+                    conflicts = self.ip_tree.get_conflicts()
+                    ip_tree_updated = False
+                    for conflict in conflicts.values():
+                        if len(conflict) > 1:
+                            auto_resolved_ip = self.try_semver_resolve_conflict(conflict)
+                            if not auto_resolved_ip is None:
+                                self.ip_tree = self.ip_tree.normalize_dependency(auto_resolved_ip)
+                                ip_tree_updated = True
+            else:
+                print(tcolors.ERROR + "Semantic versioning resolution is enabled but python package 'semver' is not "
```

I think the user requested semantic versioning at this point (`use_semver_resolution`), but he doesn't fulfill the requirements (missing package)  so the most sensible thing is to fail after printing the error instead of trying to continue withouth semver.

---

Done